presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -106,12 +118,21 @@
 
     protected final ConnectionFactory connectionFactory;
     protected final String identifierQuote;
+    protected final boolean caseInsensitiveNameMatching;
+    protected final Cache<SchemasCacheKey, Map<String, String>> remoteSchemaNames;
```

Comment:
cache key should contain identity

Reply:
i don't think contents of the cache depend on the identity
while i don't have anything fundamentally against, this can limit cache effectiveness

Reply:
> i don't think contents of the cache depend on the identity

Why do you think this way? Currently depending on extra-credentials we can connect external service as different user. So not using Identity here would be a bug. What am I missing here?

Reply:
If connecting as a different user you see different schemas and tables, then i think this is a corner case (theoretical only?).
In such a case cache should be disabled.

Reply:
done